crates/mem-pool/src/pool.rs

```diff
@@ -1275,6 +1301,63 @@ impl MemPool {
         let tx_receipt =
             TxReceipt::build_receipt(tx.witness_hash().into(), run_result, merkle_state);
 
+        // fan-out to readonly mem block
+        if let Some(handler) = &self.mem_pool_publish_service {
+            handler.new_tx(tx)
+        }
+
         Ok(tx_receipt)
     }
+
+    // Only **ReadOnly** node needs this.
+    // Refresh mem block with those params.
+    // This function returns Ok(Some(block_number)), if refresh is successful.
+    // Or returns Ok(None) if current tip has not synced yet.
+    pub(crate) fn refresh_mem_block(
+        &mut self,
+        block_info: BlockInfo,
+        withdrawals: Vec<WithdrawalRequest>,
+        deposits: Vec<DepositInfo>,
+    ) -> Result<Option<u64>> {
+        let tip_block = {
+            let db = self.store.begin_transaction();
+            db.get_last_valid_tip_block()?
+        };
+        let current_tip_number = tip_block.raw().number().unpack();
+        let next_block_number = block_info.number().unpack();
+        if next_block_number != current_tip_number + 1 {
+            return Ok(None);
+        }
+
+        let post_merkle_state = tip_block.raw().post_account();
+        let mem_block = MemBlock::new(block_info, post_merkle_state);
+        self.mem_block = mem_block;
+
+        let db = self.store.begin_transaction();
+
+        self.finalize_withdrawals(&db, withdrawals)?;
+        self.finalize_deposits(&db, deposits)?;
+
```

Comment:
In this line, when a new tip is synced, mem pool will try to collect deposits from CKB chain.
https://github.com/nervosnetwork/godwoken/blob/develop/crates/mem-pool/src/pool.rs#L816
We should change this line to:

```
if node_mode != NodeMode::ReadOnly {
  ... original logic
} else {
  self.pending_deposits.clear() // clear pending deposits
}
```


And this line, after synced new block, mem pool will try to write txs & withdrawals to current state.
https://github.com/nervosnetwork/godwoken/blob/develop/crates/mem-pool/src/pool.rs#L861

```
if node_mode != NodeMode::ReadOnly {
  self.prepare_next_mem_block(&db, withdrawals_iter, txs_iter)?;
}
```